tests/test_pasqal.py

```diff
@@ -77,40 +82,38 @@ def __init__(
         runs=10,
         variables={"t": 100, "qubits": {"q0": 1, "q1": 2, "q2": 4, "q3": 3}},
         result={"00": 5, "11": 5},
+        status=JobStatus.DONE.name,
     ) -> None:
         self.runs = runs
         self.variables = variables
         self.result = result
         self.id = str(np.random.randint(10000))
+        self.status = status
 
 
-@pytest.fixture
-def mock_job():
-    return _MockJob()
-
-
-@pytest.fixture
-def mock_batch(mock_job, seq):
-    seq_ = copy.deepcopy(seq)
-    seq_.declare_channel("rydberg_global", "rydberg_global")
-    seq_.measure()
-
-    @dataclasses.dataclass
-    class MockBatch:
-        id = "abcd"
-        status = "DONE"
-        ordered_jobs = [
-            mock_job,
+@dataclasses.dataclass
+class MockBatch:
+    id = "abcd"
+    status: str = "DONE"
+    ordered_jobs: list[_MockJob] = dataclasses.field(
+        default_factory=lambda: [
+            _MockJob(),
```

Comment:
If I get it right, the changes were needed so that you gained access to take the `MockBatch` definition outside of the fixture?

Reply:
Yeah the goal was to be able to customize the MockBatch, e.g. to set its status or results. This enables testing for the "partial results" case